test/pairwise.jl

```diff
@@ -23,24 +29,35 @@ arbitrary_fun(x, y) = cor(x, y)
         @test res == res2 == [f(xi, yi) for xi in x, yi in y]
 
         res = pairwise(f, y, z)
-        @test res isa Matrix{Float32}
-        res2 = zeros(Float32, size(res))
-        @test pairwise!(f, res2, y, z) === res2
-        @test res == res2 == [f(yi, zi) for yi in y, zi in z]
+        if isrankcor
+            @test res isa Matrix{Float64}
+            @test res == [f(yi, zi) for yi in y, zi in z]
+        else
+            @test res isa Matrix{Float32}
+            res2 = zeros(Float32, size(res))
+            @test pairwise!(f, res2, y, z) === res2
+            @test res == res2 == [f(yi, zi) for yi in y, zi in z]
+        end
 
         res = pairwise(f, Any[[1.0, 2.0, 3.0], [1.0f0, 3.0f0, 10.5f0]])
         @test res isa Matrix{Float64}
         res2 = zeros(AbstractFloat, size(res))
         @test pairwise!(f, res2, Any[[1.0, 2.0, 3.0], [1.0f0, 3.0f0, 10.5f0]]) === res2
-        @test res == res2 ==
-            [f(xi, yi) for xi in ([1.0, 2.0, 3.0], [1.0f0, 3.0f0, 10.5f0]),
-                           yi in ([1.0, 2.0, 3.0], [1.0f0, 3.0f0, 10.5f0])]
+
+        res3 = [f(xi, yi) for xi in ([1.0, 2.0, 3.0], [1.0f0, 3.0f0, 10.5f0]),
+                yi in ([1.0, 2.0, 3.0], [1.0f0, 3.0f0, 10.5f0])]
+        @test res == res2
+        @test all(isapprox.(res2, res3))
+
         @test res isa Matrix{Float64}
 
         @inferred pairwise(f, x, y)
-
-        @test_throws Union{ArgumentError,MethodError} pairwise(f, [Int[]], [Int[]])
-        @test_throws Union{ArgumentError,MethodError} pairwise!(f, zeros(1, 1), [Int[]], [Int[]])
+        if throwsforzerolengthinput
+            # @test_throws Union{ArgumentError,MethodError} pairwise(f, [Int[]], [Int[]])
+            @test_throws CompositeException pairwise(f, [Int[]], [Int[]])
+            #@test_throws Union{ArgumentError,MethodError} pairwise!(f, zeros(1, 1), [Int[]], [Int[]])
```

Comment:
Is this commented out intentionally? There are a few other tests that are commented out.

Reply:
The commented-out test at line 56 was replaced by the test at line 57 (necessary since errors in threaded code burble up as `CompositeException`). So I will delete the commented out test at 56 and similar occurrences.

Reply:
Fixed in commit 1a4d064